src/app/pages/BuyCrypto/BuyCrypto.tsx

```diff
@@ -1,16 +1,113 @@
-import React, { FC, useState } from 'react';
+import React, { FC, ReactNode, Suspense, useMemo, useState } from 'react';
 
+import classNames from 'clsx';
+
+import Spinner from 'app/atoms/Spinner';
 import Stepper from 'app/atoms/Stepper';
+import { useAppEnv } from 'app/env';
+import ErrorBoundary from 'app/ErrorBoundary';
 import PageLayout from 'app/layouts/PageLayout';
 import ApproveStep from 'app/pages/BuyCrypto/steps/ApproveStep';
 import ExchangeStep from 'app/pages/BuyCrypto/steps/ExchangeStep';
 import InitialStep from 'app/pages/BuyCrypto/steps/InitialStep';
 import { ExchangeDataInterface } from 'lib/exolix-api';
 import { T, t } from 'lib/i18n/react';
 import { useAccount, useNetwork, useStorage } from 'lib/temple/front';
-import { Redirect } from 'lib/woozie';
+import { Redirect, Link } from 'lib/woozie';
+
+import { BuyCryptoSelectors } from './BuyCypto.selectors';
+import { Cryptos } from './Cryptos';
+import { Debits } from './Debits';
+import { useTabSlug } from './hooks/use-tab-slug';
+
+interface BuyCryptoProps {
+  crypto?: boolean;
+}
+
+const BuyCrypto: FC<BuyCryptoProps> = ({ crypto = false }) => {
+  const { fullPage } = useAppEnv();
+  const tabSlug = useTabSlug();
+
+  const tabs = useMemo<
+    {
+      slug: string;
+      title: string;
+      Component: FC;
+      testID: string;
+    }[]
+  >(() => {
+    return [
+      {
+        slug: 'crypto',
+        title: t('topUpCrypto'),
+        Component: Cryptos,
+        testID: BuyCryptoSelectors.Crypto
+      },
+      {
+        slug: 'debit',
+        title: t('topUpDebit'),
+        Component: Debits,
+        testID: BuyCryptoSelectors.Debit
+      }
+    ];
+  }, []);
+
+  const { slug, Component } = useMemo(() => {
+    const tab = tabSlug ? tabs.find(currentTab => currentTab.slug === tabSlug) : null;
+    return tab ?? tabs[0];
+  }, [tabSlug, tabs]);
+
+  if (crypto) {
+    return <BuyCryptoOld />;
+  }
```

Comment:
What is the reason of using this `crypto` flag?  
Why don't just split into 2 separate components?